### ISSUE TITLE
[DataGrid] DataGridPro type check fails at build with themeAugmentation

### ISSUE THREAD

**Issue by AntonWiliot** [mui/mui-x#11394]
### The problem in depth 🔍

After we switched the type definitions in our MUI theme settings from "@mui/x-data-grid/themeAugmentation" to "@mui/x-data-grid-pro/themeAugmentation" - `import type {} from "@mui/x-data-grid-pro/themeAugmentation";` - the next typescript errors appeared:
```
[build:types] node_modules/@mui/x-data-grid-pro/models/dataGridProProps.d.ts(177,41): error TS2344: Type '"rowsScrollEnd"' does not satisfy the constraint 'keyof GridEventLookup'.
[build:types] node_modules/@mui/x-data-grid-pro/models/dataGridProProps.d.ts(214,42): error TS2344: Type '"rowOrderChange"' does not satisfy the constraint 'keyof GridEventLookup'.
[build:types] node_modules/@mui/x-data-grid-pro/models/dataGridProProps.d.ts(221,37): error TS2344: Type '"fetchRows"' does not satisfy the constraint 'keyof GridEventLookup'.
```
The issue appears at build stage when tsc command is run. The problem is observed for Pro version only, Community version has no issues. Please could you check is it the issue in the library types or may be the problem in our configuration (envinfo details are added).

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>
System:
    OS: macOS 14.0
  Binaries:
    Node: 18.17.1 - /usr/local/bin/node
    Yarn: 1.22.18 - /usr/local/bin/yarn
    npm: 9.6.7 - /usr/local/bin/npm
  Browsers:
    Chrome: 120.0.6099.71
    Edge: Not Found
    Safari: 17.0
  npmPackages:
    @emotion/react: ^11.11.0 => 11.11.1 
    @emotion/styled: ^11.11.0 => 11.11.0 
    @mui/base:  5.0.0-beta.26 
    @mui/core-downloads-tracker:  5.14.20 
    @mui/icons-material: ^5.14.0 => 5.14.19 
    @mui/lab: ^5.0.0-alpha.140 => 5.0.0-alpha.155 
    @mui/material: ^5.14.0 => 5.14.20 
    @mui/private-theming:  5.14.20 
    @mui/styled-engine:  5.14.20 
    @mui/styled-engine-sc: ^5.14.0 => 5.14.12 
    @mui/system:  5.14.20 
    @mui/types:  7.2.10 
    @mui/utils:  5.14.20 
    @mui/x-data-grid:  6.18.4 
    @mui/x-data-grid-pro: ^6.18.0 => 6.18.4 
    @mui/x-license-pro:  6.10.2 
    @types/react: ^17.0.19 => 17.0.71 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 
    styled-components: ^5.3.9 => 5.3.11 
    typescript: ^5.3.0 => 5.3.3 
</details>

**Search keywords**: datagridpro typescript error
**Order ID**: 71637

**Comment by michelengelen** [comment]
Hey @AntonWiliot I guess you are still using `GridEventLookup`, but the data grid pro augmentation uses `GridEventLookupPro`. 
Could you check if you import and use `GridEventLookup` in your implementation? In that case changing this to `GridEventLookupPro` should be sufficient to satisfy the key constraint!
Thanks!

**Comment by AntonWiliot** [comment]
> I guess you are still using `GridEventLookup`, but the data grid pro augmentation uses `GridEventLookupPro`. Could you check if you import and use `GridEventLookup` in your implementation? In that case changing this to `GridEventLookupPro` should be sufficient to satisfy the key constraint! Thanks!

No, we don't use explicit imports of `GridEventLookup`. Only `type {} from "@mui/x-data-grid-pro/themeAugmentation"` is imported in MUI theme settings. The error appears just after change @mui/x-data-grid to @mui/x-data-grid-pro, without any other changes. The application works, only type check fails. The issue looks to be in type definitions inside the pro package

**Comment by michelengelen** [comment]
@AntonWiliot I tried, but cannot reproduce the issue. Could you provide us with a minimal reproduction example and include your typescript config?

@cherniavskii The only thing that I could find (that did not make much sense to me - or maybe i am missing something) was this part in the `typeOverloads/modules.ts` file:

**for Prop**
https://github.com/mui/mui-x/blob/c85d38b5b8975df15a9f5a2264c2e59f8464ed13/packages/grid/x-data-grid-pro/src/typeOverloads/modules.ts#L64-L76

**for Premium**
https://github.com/mui/mui-x/blob/c85d38b5b8975df15a9f5a2264c2e59f8464ed13/packages/grid/x-data-grid-premium/src/typeOverloads/modules.ts#L91-L109

Shouldn't this be ...
```typescript
declare module '@mui/x-data-grid-pro' {
...
```
... for Pro, and ...
```typescript
declare module '@mui/x-data-grid-premium' {
...
```
... for Premium?

**Comment by AntonWiliot** [comment]
> I tried, but cannot reproduce the issue. Could you provide us with a minimal reproduction example and include your typescript config?

Minimal example of file where the problem occurs
```
import { ThemeOptions, createTheme } from "@mui/material/styles";
import type {} from "@mui/x-data-grid-pro/themeAugmentation";
const themeSettingsMUI: ThemeOptions = {};
export const themeMUI = createTheme(themeSettingsMUI);
```

Typescript config
```
{
  "compilerOptions": {
    "esModuleInterop": true,
    "module": "esnext",
    "allowSyntheticDefaultImports": true,
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "lib": [
      "dom",
      "es5",
      "scripthost",
      "es2015",
      "es2015.promise",
      "es2016.array.include",
      "es2018"
    ],
    "declaration": true,
    "emitDeclarationOnly": true,
     "jsx": "react-jsx",
    "declarationDir": "dist"
  }
 "include": ["src/**/*"],
  "exclude": ["src/**/*.test*"]
}
```

**Comment by cherniavskii** [comment]
Hi @AntonWiliot 
Thanks for the minimal example.
Unfortunately, I cannot reproduce the issue you describe - please take a look if you can reproduce it in this repo: https://github.com/cherniavskii/data-grid-pro-theme-augmentation

**Comment by AntonWiliot** [comment]
I've cloned your repository, installed packages, run `tsc` and got all the errors mentioned above:

```
node_modules/@mui/x-data-grid-pro/models/dataGridProProps.d.ts:177:41 - error TS2344: Type '"rowsScrollEnd"' does not satisfy the constraint 'keyof GridEventLookup'.
onRowsScrollEnd?: GridEventListener<'rowsScrollEnd'>;

node_modules/@mui/x-data-grid-pro/models/dataGridProProps.d.ts:214:42 - error TS2344: Type '"rowOrderChange"' does not satisfy the constraint 'keyof GridEventLookup'.
onRowOrderChange?: GridEventListener<'rowOrderChange'>;

node_modules/@mui/x-data-grid-pro/models/dataGridProProps.d.ts:221:37 - error TS2344: Type '"fetchRows"' does not satisfy the constraint 'keyof GridEventLookup'.
onFetchRows?: GridEventListener<'fetchRows'>;
```

So the error is recreated in the reported minimal configuration

**Comment by cherniavskii** [comment]
Thanks, I've upgraded `typescript` and can reproduce the issue now.
Is there a reason you don't have `skipLibCheck` enabled in `tsconfig.json`?

**Comment by AntonWiliot** [comment]
We'd prefer to see the issues in the packages we use. Sure if something not critical blocks our deployment `skipLibCheck` option may be added. Currently we temporarily use community DataGrid type augmentation until Pro types will be fixed

**Comment by michelengelen** [comment]
@cherniavskii i'll add this to the board, ok?
